src/core/ddsc/src/dds_topic.c

```diff
@@ -596,6 +596,7 @@ dds_entity_t dds_create_topic (dds_entity_t participant, const dds_topic_descrip
 #ifdef DDS_HAS_SHM
   st->c.iox_size = desc->m_size;
 #endif
+  st->c.fixed_size = (st->c.fixed_size || (desc->m_flagset & DDS_TOPIC_FIXED_SIZE)) ? 1u : 0u;
```

Comment:
Instead of `ddsi_sertype_init` + setting the `fixed_size` flag afterward, it'd probably make more sense to use `ddsi_sertype_init_flags`. The `iox_size` bit still needs to be integrated as well, but let's not add more "fix ups" after `ddsi_sertype_init` than necessary!